doc/storage-encryption.md

```diff
@@ -0,0 +1,115 @@
+Storage Encryption
+------------------
+
+A PKCS#11 token uses a storage cmponent to save keys.
+In the case of a software token, it is a good idea to encrypt sensible
+information in the storage so that keys cannot be easily stolen or tampered
+with.
+
+Kryoptic adopts a flexible and agile encryption system that allows for future
+changes while maintaining reasonable comaptibility.
+
+The aim of the v1 scheme identified here is to provide reasonable security
+while allowing simplified operations.
+
+The token allows for two PINs to be used the SO PIN and the USER PIN.
+The SO PIN is considered a recvoery PIN. Both PINs allow for access to the
+key material.
+
+The PINs are used in conjunction with PBKDF2 to derive a KEK that is used to
+decrypt the storage Encryption key. The Encryption key is versioned and is used
+to encrypt each secret attribute individually by using a key derivation step to
+obtain an object specific key. The Key Derivation algorithms and paramters are
+felxibly defined by wrapping the keys and encrypted values using PKCS#5 like
+ASN.1 structures.
+
+This OID has been defined to identify the HKDF-Expand Key Derivation function:
+1.3.6.1.4.1.2312.20.1.1
+
+KKBPS - Kryoptic Key Based Protection Scheme v1
```

Comment:
Every time I read the shortcut `kbps` in lower case I read it as unit of transfer speed kB/s, which is a bit confusing. But I do not have better naming suggestion ...

Reply:
I know, there isn't much we can do